synapse/tests/test_common.py

```diff
@@ -439,6 +439,32 @@ async def foo():
 
         self.eq(123, await s_common.wait_for(footask, timeout=-1))
 
+    async def test_waitgenr(self):
+
+        async def one_item_genr():
+            yield 'item'
+
+        items = []
+        async for retn in s_common.waitgenr(one_item_genr(), timeout=1.0):
+            items.append(retn)
+
+        self.len(1, items)
+        self.eq((True, 'item'), items[0])
+
+        async def slow_genr():
+            yield 'item1'
+            await asyncio.sleep(2)
+            yield 'item2'
+
+        items = []
+        async for retn in s_common.waitgenr(slow_genr(), timeout=0.1):
+            items.append(retn)
```

Comment:
fyi, there's a handy utility for this, `test.utils.alst()`.

Reply:
Nice! This might be one of those tests that get dropped, @invisig0th also added a `test_waitgenr` in his PR too.

Reply:
Did we mostly implement the same test or did i forget to test a case we need?

Reply:
Very similar, I've dropped my func and back ported a `StopAsyncIteration` test to yours. Will be up in the next commit.